### ISSUE TITLE
Provide a way to run IDE with separate backend and frontend processes and have the plugin installed on the frontend side

### ISSUE THREAD

**Issue by chashnikov** [JetBrains/intellij-platform-gradle-plugin#1563]
### Describe the need of your request

If you want to develop a plugin that should work in the frontend process (JetBrains Client), it'll be good to have a way to debug such a plugin without manually creating its distribution and installing it.

### Proposed solution

In #1562 a task which runs the backend part and the frontend part of the IDE in separate processes is introduced. By default, it includes the developed plugin in the backend process. Until we can automate determine whether the plugin is supposed to work on the backend side, on the frontend side or both, let's provide a configuration option to manually specify which process should have the plugin.

**Comment by AlexPl292** [comment]
JFI, IdeaVim plugin works only on the client side, so this would be really helpful for us.

**Comment by citizenmatt** [comment]
Is this planned for 2.0? It's the only reason IdeaVim would use split mode, so would be great if it can be in added sooner rather than later. Thanks!

**Comment by chashnikov** [comment]
@citizenmatt I'm working on this, hope to have it done soon.

**Comment by citizenmatt** [comment]
Cool. Thank you!

**Comment by chashnikov** [comment]
This should be fixed in [2.0.0-beta2](https://github.com/JetBrains/intellij-platform-gradle-plugin/releases/tag/v2.0.0-beta2). Use the following configuration in build.gradle.kts:
```
intellijPlatform {
    splitMode = true
    splitModeTarget = SplitModeAware.SplitModeTarget.FRONTEND
}
```

**Comment by chashnikov** [comment]
It appears that currently this doesn't work on macOS due to customized layout of IDE installation used by intellij-platform-gradle-plugin. I'll fix this on IDE side [RDCT-1375](https://youtrack.jetbrains.com/issue/RDCT-1375).

**Comment by chashnikov** [comment]
The problem with installing plugin on macOS should be fixed if the next 2024.2 EAP build of the IDEs.
